### PR TITLE
Update safe integration test

### PR DESCRIPTION
**Pull request by JohnGuilding**
- Pulls latest Safe 7579 changes
- Updates Safe integration tests
- Updates Safe unit tests
- Runs multi-account tests in CI (default, Safe, Kernel)
- Runs simulate tests in CI (Checks against 4337 storage access rules)

### CODE REVIEW

File: script/DeploySafeRecovery.s.sol [hunk 1]
```diff
@@ -2,25 +2,68 @@
...
+
+// 1. `source .env`
+// 2. `forge script --chain sepolia script/DeploySafeRecovery.s.sol:DeploySafeRecovery_Script
+// --rpc-url $BASE_SEPOLIA_RPC_URL --broadcast -vvvv`
+contract DeploySafeRecovery_Script is Script {
     function run() public {
-        address verifier = 0xEdC642bbaD91E21cCE6cd436Fdc6F040FD0fF998;
-        address dkimRegistry = 0xC83256CCf7B94d310e49edA05077899ca036eb78;
-        address emailAuthImpl = 0x1C76Aa365c17B40c7E944DcCdE4dC6e6D2A7b748;
+        bytes32 salt = bytes32(uint256(0));
```
Comment (SoraSuegami) [inline script/DeploySafeRecovery.s.sol]: Why is the salt fixed here?

File: script/DeploySafeRecovery.s.sol [hunk 2]
```diff
@@ -2,25 +2,68 @@
...
+            new EmailRecoveryManager(verifier, dkimRegistry, emailAuthImpl, emailRecoveryHandler)
         );
+        address emailRecoveryModule =
+            address(new UniversalEmailRecoveryModule(emailRecoveryManager));
+
+        EmailRecoveryManager(emailRecoveryManager).initialize(emailRecoveryModule);
+
+        address safe7579 = address(new Safe7579{ salt: salt }());
+        address safe7579Launchpad =
+            address(new Safe7579Launchpad{ salt: salt }(entryPoint, registry));
```
Comment (SoraSuegami) [inline script/DeploySafeRecovery.s.sol]: Where/when does this safe account install necessary modules?

Comment (JohnGuilding) [reply script/DeploySafeRecovery.s.sol]: For this script, it is to deploy the contracts for the frontend demo. The module is installed via the frontend.

File: script/DeploySafeRecovery.s.sol (hunk 1 above)
Comment (JohnGuilding) [reply script/DeploySafeRecovery.s.sol]: Yeah maybe that is inconsistent with the other code. Deployed other contracts with factory which uses salts.

The code to deploy the launchpad and safe7579 contract had a salt like this, so I guess I copied it over

**Review by SoraSuegami — APPROVED** [review]

**Pull request merged**